api/desecapi/authentication.py

```diff
@@ -156,9 +156,17 @@ class AuthenticatedBasicUserActionAuthentication(BaseAuthentication):
     """
     def authenticate(self, request):
         view = request.parser_context['view']
-        serializer = AuthenticatedBasicUserActionSerializer(data={}, context=view.get_serializer_context())
+        return self.authenticate_credentials(view.get_serializer_context())
+
+    def authenticate_credentials(self, context):
+        serializer = AuthenticatedBasicUserActionSerializer(data={}, context=context)
         serializer.is_valid(raise_exception=True)
-        return serializer.validated_data['user'], None
+
+        user = serializer.validated_data['user']
+        if user.is_active == False:  # don't catch None here!
+            raise exceptions.AuthenticationFailed('User inactive.')
```

Comment:
All out-of-the-box Django authenticators have this check.

Reply:
Good point, but this PR also introduces a permission for this purpose, no?

Reply:
No. This authenticator fails if `user.is_active` is exactly `False`, i.e. it does not fail when activation is pending (`user.is_active is None`). The `IsActiveUser` permission on the other hand requires `user.is_active` to be truthy, i.e. active *and* activated.

A user that is not active has no business with the API whatsoever. A user which is pending or fully active has a different set of permissible activities (e.g. pending: performing the activation action but not DNS updates), so that's a permission.

Reply:
alright let's add some more docs and keep it this way

Reply:
done